### ISSUE TITLE
HTTP Health check add Response HTTP codes option

### ISSUE THREAD

**Issue by 123BLiN** [hashicorp/consul#2294]
Hello, 

according to docs:
> any 2xx code is considered passing, a 429 Too Many Requests is a warning, and anything else is a failure. 

It would be nice to leave it as default behaviour, but optionally add ability to set response HTTP codes that also should be considered as good.
For exmaple: 302, 405, 403

**Comment by laymain** [comment]
By the way, a 302 response is not considered as a failure, redirection is followed

**Comment by laymain** [comment]
@RiRa12621 I agree with you, that's a problem. I have been confronted to a case where a 302 response was a service failure so I had to deploy my own script to check the HTTP response code

**Comment by slackpad** [comment]
We should definitely update the docs as well to make the 3xx behavior more clear.

**Comment by pearkes** [comment]
@jefferai just brought this up as it pertains to Vault. The current usage of 429 as a warning status conflicts with the common interpretation of [RFC 6585](https://tools.ietf.org/html/rfc6585#section-4).

Given many applications likely depend on this current default, and we don't want to incur a high cost to all users of it, we'd likely add the ability to configure it as this issue states, add an additional more "accurate" status code (429 + something else) and then eventually deprecate 429 as an option. Users dealing with legacy applications they do not want to modify could just retain 429 in their configuration for the future.
